File: frontend/src/types/dispute.ts
export type DisputeStatus = 'OPEN' | 'IN_REVIEW' | 'RESOLVED' | 'REJECTED'

export interface Dispute {
  disputeId: number
  customerId: number
  txnId: number | null
  reason: string
  status: DisputeStatus
  resolutionNotes: string | null
  raisedAt: string
  resolvedAt: string | null
}

export interface RaiseDisputeRequest {
  txnId?: number
  reason: string
}

export interface ResolveDisputeRequest {
  status: DisputeStatus
  resolutionNotes?: string
}

File: backend/src/main/java/com/smartbank/manager/dispute/Dispute.java
package com.smartbank.manager.dispute;

import com.smartbank.manager.customer.Customer;
import com.smartbank.manager.transaction.Transaction;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import java.time.Instant;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "disputes")
@Getter
@Setter
@NoArgsConstructor
public class Dispute {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "dispute_id")
    private Long disputeId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "customer_id", nullable = false)
    private Customer customer;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "txn_id")
    private Transaction transaction;

    @Column(name = "reason", nullable = false)
    private String reason;

    @Enumerated(EnumType.STRING)
    @Column(name = "status", nullable = false)
    private DisputeStatus status = DisputeStatus.OPEN;

    @Column(name = "resolution_notes")
    private String resolutionNotes;

    @Column(name = "raised_at", nullable = false, updatable = false)
    private Instant raisedAt = Instant.now();

    @Column(name = "resolved_at")
    private Instant resolvedAt;
}
